garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stool.lua

```diff
@@ -155,7 +155,9 @@ for _, val in ipairs( file.Find( SWEP.Folder .. "/stools/*.lua", "LUA" ) ) do
 
 	TOOL:CreateConVars()
 
-	SWEP.Tool[ toolmode ] = TOOL
+	if ( hook.Run( "PreRegisterTOOL", t, name ) != false ) then

```

Comment:
Either `name` not `t` exist in this context, so your hook is being called with 2 `nil` arguments.